File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        for (int i =0; i<s.length(); i++){
            int ans = -1;
            for (int j=i-1; j>=0; j--){
                if(s.charAt(j) == s.charAt(i)){
                    ans = i-j;
                    break;
                }
            }answer[i] = ans;
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    arr=[]
    strr=[]
    arr = list(s)
    #print(arr)

    for i in arr :
        if i not in strr :
            answer.append(-1)
            strr.append(i)
        else :
            ind=[]
            for j in range(len(strr)) :
                if strr[j]==i :
                    ind.append(j)

            strr.append(i)
            #print(strr,ind)
            answer.append(len(strr)-max(ind)-1)

    return answer